src/caip-chain-id.ts

```diff
@@ -0,0 +1,67 @@
+import { is, pattern, string, Struct } from 'superstruct';
+
+import { assert } from './assert';
+
+export type CaipChainId = `${string}:${string}`;
+
+const CAIP2_REGEX = /^([-a-z0-9]{3,8}):([-_a-zA-Z0-9]{1,32})$/u;
+
+export const CaipChainIdStruct = pattern(string(), CAIP2_REGEX) as Struct<
+  CaipChainId,
+  null
+>;
+
+export type ParsedCaipChainId = {
+  namespace: string;
+  reference: string;
+};
+
+/**
+ * Check if a string is a valid caip chain id string.
+ *
+ * @param value - The value to check.
+ * @returns Whether the value is a valid caip chain id string.
+ */
+export function isCaipChainIdString(value: unknown): value is CaipChainId {
```

Comment:
It would be good to tighten the terminology here. Currently in this file you have:

- CaipChainId
- CaipChainIdString
- ParsedCaipChainId

So I have two questions:

- When we refer to a "caipChainId" in the code, what do we expect? Do we expect a string or do we expect an object?
- How often will we use the parsed chain ID vs. the unparsed chain ID?

Reply:
Right now, we expect a string. ParsedCaipChainId is used only in core's `controller-utils`. See [this file](https://github.com/MetaMask/core/pull/1489/files#diff-09bd0506f5dc77f407d068e86b89713e515a71cb587dbfccfae07377e23894bfR1) in the other PR. The linked file provides helpers that extensions uses to pull the reference out of the caipChainId into more usable forms.

My idea was to provide the common logic needed for CAIP-2 Chain IDs here in `@metamask/utils` and to provide Eth specific helpers in `@metamask/controller-utils`